### PR TITLE
DOC: missing submodule init information

### PR DESCRIPTION
**Pull request by tupui**
Partially adresses #14048

Add information about git submodule in the documentation (especially quickstarters guides).

### CODE REVIEW

**Comment by tupui** [comment]
@mdhaber so you know, this was a source or hurdle today for the sprint.

File: doc/source/dev/contributor/development_workflow.rst [hunk 1]
```diff
@@ -38,6 +38,12 @@ SciPy.
    a "pull request" asking SciPy to "pull" your changes from your fork into
    the official repository.
 
+#. Initialize git submodules::
+
+      git submodule update --init
+
+   This fetch and update any submodules that SciPy needs (such as `Boost`).
```
Comment (mdhaber) [inline doc/source/dev/contributor/development_workflow.rst:45]:
```suggestion
   This fetches and updates any submodules that SciPy needs (such as `Boost`).
```

Comment (tupui) [reply doc/source/dev/contributor/development_workflow.rst:45]: Great, thanks Matt!

**Comment by mdhaber** [comment]
Renders correctly in all development environment setup guides except the old mac and linux guides [here](https://30953-1460385-gh.circle-artifacts.com/0/html-scipyorg/building/index.html#building). We should probably just remove those, but those can be a separate PR.

**Comment by tupui** [comment]
I could make a PR to remove them. Although there might still be valuable info no?

**Pull request merged**